src/App.tsx

```diff
@@ -1,34 +1,108 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { FilterStatus } from './types/FilterStatus';
+import { getTodos, getUser } from './api';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
-export const App: React.FC = () => {
+export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [filter, setFilter] = useState<FilterStatus>(FilterStatus.All);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [modalLoading, setModalLoading] = useState(false);
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      try {
+        const loadedTodos = await getTodos();
+
+        setTodos(loadedTodos);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    loadTodos();
+  }, []);
+
+  const handleCloseModal = useCallback(() => {
+    setSelectedTodo(null);
+    setSelectedUser(null);
+  }, []);
+
+  const handleShowTodo = useCallback(
```

Comment:
you broke SOLID 
your function name showTodo and i expect that it is only responsible for display but there is more logic going on inside
maybe it is worth renaming the functions or separating the logic